detox/android/detox/src/full/java/com/wix/detox/reactnative/ui/UIExtensions.kt

```diff
@@ -0,0 +1,39 @@
+package com.wix.detox.reactnative.ui
+
+import android.view.View
+import android.widget.TextView
+import com.wix.detox.common.forEachChild
+import com.wix.detox.reactnative.utils.isReactNativeObject
+
+fun View.getAccessibilityLabel(
+    isReactNativeObjectFn: (Any) -> Boolean = { isReactNativeObject(it) }
+): CharSequence? =
+    if (isReactNativeObjectFn(this)) {
+        val subLabels = collectChildAccessibilityLabels(this)
+        if (subLabels.isEmpty()) null else subLabels.joinToString(" ")
+    } else {
+        getRawAccessibilityLabel(this)
+    }
+
+private fun collectChildAccessibilityLabels(
+    view: View,
+    subLabels: MutableList<CharSequence> = mutableListOf(),
+): List<CharSequence>{
+    getRawAccessibilityLabel(view).let { rawLabel ->
+        if (rawLabel != null) {
+            subLabels.add(rawLabel)
+        } else {
+            view.forEachChild { child ->
+                collectChildAccessibilityLabels(child, subLabels)
+            }
+        }
+    }
+    return subLabels
+}
```

Comment:
You could avoid this recursion using a queue:
```kotlin
private fun collectChildAccessibilityLabels(
    view: View,
): List<CharSequence> {
    val labels = mutableListOf<CharSequence>()
    val queue = mutableListOf<View>()

    queue.add(view)

    while (queue.isNotEmpty()) {
        val currentView = queue.removeAt(0)
        getRawAccessibilityLabel(currentView)?.let { rawLabel ->
            labels.add(rawLabel)
        } ?: run {
            view.forEachChild { child ->
                queue.add(child)
            }
        }
    }

    return labels
}

```

Reply:
I liked the idea of traversing the view hierarchy as a general method.